### PR TITLE
Rust: add bench for producer creation/dump

### PR DESCRIPTION
**Pull request by jmillan**

### CODE REVIEW

**Comment by jmillan** [comment]
@nazar-pc, I've added some types to the prelude as I'm doing things directly using Rust types. Is there anything against that or a better way to do it?

BTW, most part of the bench code is copied from the producer test.

**Review by ibc — COMMENTED** [review]
Just in Rust? There are built-in benchmark utils in Node.

File: rust/benches/producer.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        });
+
+        group.bench_function("create", |b| {
+            b.iter(|| {
+                let _ = futures_lite::future::block_on(async {
+                    transport_1
+                        .produce(audio_producer_options())
+                        .await
+                        .expect("Failed to produce audio")
+                });
```
Comment (nazar-pc) [inline rust/benches/producer.rs:145]: You are not only testing creation here, you also trigger destruction by dropping the value here. Destruction is async though, so I guess the bench is fine for comparison reasons.

**Review by nazar-pc — APPROVED** [review]
There is a better way to run async benches, but this will do for now.

**Comment by nazar-pc** [comment]
> @nazar-pc, I've added some types to the prelude as I'm doing things directly using Rust types. Is there anything against that or a better way to do it?

Prelude should contain most popular types. Some more exotic ones have to be imported separarely and explicitly. What is "most popular" is, of course, subjective. In Rust `Vec` doesn't need to be imported explicitly, but `VecDeque` does for example.

**Comment by jmillan** [comment]
> Just in Rust? There are built-in benchmark utils in Node.

Any pointer to those?

**Comment by jmillan** [comment]
> > Just in Rust? There are built-in benchmark utils in Node.
> 
> Any pointer to those?

I see https://nodejs.org/api/perf_hooks.html.

I'll add the same ones we have for Rust.

**Comment by jmillan** [comment]
> Just in Rust? There are built-in benchmark utils in Node.

@ibc, how do you propose to do this?

We have no benches in TS so far. Should it be done within the jest tests? Should we have a `benches` directory and build our own simple benching suite? 

I've just added a bench in Rust which has a built in bench mechanism run by `cargo bench`, but we don't have such thing in TS. If we want to have this in Node we should think about it.

Is this really a blocker for this PR?

**Comment by ibc** [comment]
> > Just in Rust? There are built-in benchmark utils in Node.
> 
> @ibc, how do you propose to do this?

Honestly I just asked about Node :)

> We have no benches in TS so far. Should it be done within the jest tests? Should we have a `benches` directory and build our own simple benching suite?
> 
> I've just added a bench in Rust which has a built in bench mechanism run by `cargo bench`, but we don't have such thing in TS. If we want to have this in Node we should think about it.
> 
> Is this really a blocker for this PR?

No. We should add such a Node benchmark stuff separately after flatbuffers is merged.

**Review by ibc — APPROVED** [review]
`cargo clippy` failing.

**Comment by jmillan** [comment]
> Honestly I just asked about Node :)

That was not how I read it :-)

**Comment by ibc** [comment]
> > Honestly I just asked about Node :)
> 
> That was not how I read it :-)

Because I didn't realize that, indeed, we don't have a proper benchmark mechanism/place for mediasoup Node.

**Pull request merged**